sos/report/plugins/openshift.py

```diff
@@ -7,151 +7,369 @@
+            if _res['status'] == 0:
+                for _res_name in _res['output'].splitlines()[1:]:
+                    self.add_cmd_output(
+                        "oc describe %s %s" % (resource, _res_name.split()[0]),
+                        subdir=_subdir
+                    )
+
+    def collect_from_namespace(self, namespace):
+        """Run through the collection routines for an individual namespace.
+        This collection should include all requested resources that exist
+        within that namesapce
+
+            :param namespace str:           The name of the namespace
+        """
+
+        # define the list of resources to collect
+        resources = [
+            'buildconfigs',
+            'builds',
+            'catalogsourceconfigs',
+            'catalogsources',
+            'clusterserviceversions',
+            'configmaps',
+            'daemonsets',
+            'deploymentconfigs',
+            'deployments',
+            'events',
+            'horizontalpodautoscalers',
+            'imagestreams',
+            'ingresscontrollers',
+            'ingresses',
+            'installplans',
+            'limitranges',
+            'machines',
+            'machinesets',
+            'mcoconfigs',
+            'net-attach-def',
+            'operatorgroups',
+            'operatorsources',
+            'pods',
+            'pvc',
+            'resourcequotas',
+            'routes',
+            'secrets',
+            'services',
+            'statefulsets',
+            'subscriptions'
+
+        ]
+
+        # save to namespace-specific subdirs to keep the plugin dir organized
+        subdir = "namespaces/%s" % namespace
 
+        # namespace-specific non-resource collections
+        self.add_cmd_output("oc describe namespace %s" % namespace,
+                            subdir=subdir)
+
+        for res in resources:
+            _subdir = "%s/%s" % (subdir, res)
+            _get_cmd = "%s --namespace=%s %s" % (self.oc_cmd, namespace, res)
+            # get the 'normal' output first
+            _res_out = self.collect_cmd_output(
+                _get_cmd,
+                subdir=_subdir
+            )
+
+            # then get specific detail on each instance of the resource
+            if _res_out['status'] == 0:
+                _instances = _res_out['output'].splitlines()[1:]
+                for _instance in _instances:
+                    _instance_name = _instance.split()[0]
+                    self.add_cmd_output(
+                        "%s %s -o yaml" % (_get_cmd, _instance_name),
+                        subdir=_subdir,
+                        suggest_filename="%s.yaml" % _instance_name
+                    )
+                # check for podlogs here as a slight optimization to re-running
+                # 'oc get pods' on all namespaces
+                if res == 'pods' and _instances:
+                    if self.get_option('podlogs'):
```

Comment:
Doesnt it make sense to merge the two `if` commands into one?

Reply:
Ack.